### PR TITLE
Add HTTPHeadersWrappingHeaders

### PR DESCRIPTION
**Pull request by wsanchez**
This class wraps `twisted.web.http_headers.Headers` object and exposes it as an `IHTTPHeaders` implementation.

### CODE REVIEW

File: src/klein/_headers_compat.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+from ._headers import (
+    IMutableHTTPHeaders, RawHeaders, String, headerNameAsBytes,
+    headerValueAsText, normalizeHeaderName, rawHeaderName,
+    rawHeaderNameAndValue,
+)
+
+AnyStr, Iterable, RawHeaders, String, Tuple  # Silence linter
+
+
+__all__ = ()
```
Comment (moshez) [inline src/klein/_headers_compat.py]: * Should be []
* Are we really not exporting any symbols? I feel like anything you expect to be used from outside should be here.

File: src/klein/_headers_compat.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            values = (
+                headerValueAsText(value)
+                for value in self._headers.getRawHeaders(
+                    headerNameAsBytes(name), default=()
+                )
+            )
+        else:
+            raise TypeError("name {!r} must be text or bytes".format(name))
+
+        return tuple(values)
```
Comment (moshez) [inline src/klein/_headers_compat.py]: I'm not a big fan of using tuple as "immutable list". Especially typing wise. Since the value is fresh anyway, why not return a list?

File: src/klein/_headers_compat.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+
+    # NOTE: In case Headers has different ideas about encoding text than we do,
```
Comment (moshez) [inline src/klein/_headers_compat.py]: The placing of this comment is weird. Its should probably be further up.

File: src/klein/test/test_headers_compat.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+# Silence linter
+IMutableHTTPHeaders, RawHeaders
+
+
+__all__ = ()
+
+
+
+class HTTPHeadersWrappingHeadersTests(MutableHTTPHeadersTestsMixIn, TestCase):
```
Comment (moshez) [inline src/klein/test/test_headers_compat.py]: I...any chance you can turn this into a composition rather than inheritance? This seems unnecessary to mix-in.

File: src/klein/_headers_compat.py (hunk 1 above)
Comment (wsanchez) [reply src/klein/_headers_compat.py]: Why `[]` and not `()`?

We aren't exporting any of this new API until the all of the (near-term, anyway) pieces are in place, so that we are confident that it has the correct shape before we make any commitments about compatibility.  Also the exports would be done from another file, not from a private `_foo` module.

File: src/klein/test/test_headers_compat.py (hunk 4 above)
Comment (wsanchez) [reply src/klein/test/test_headers_compat.py]: I've thought about that but I think the best way to do that is to fix trial to provide some means for composition instead of having to subclass `*TestCase`.  I'm not really ready to dive into that problem.

The biggest argument against subclassing (and especially mix-ins) is, I think, that it's stupidly hard to follow the intentions and track down the parent class, but it's all in one file, and test cases are not public API… So I think in examples like this one, it seems academically interesting but ultimately a bunch of work for negligible gain.

File: src/klein/_headers_compat.py (hunk 2 above)
Comment (wsanchez) [reply src/klein/_headers_compat.py]: Right, it's spec'ed as an `Iterable`, so a `list` should be fine.

Comment (wsanchez) [reply src/klein/_headers_compat.py]: d772564c7425ec705126b1db386ad4abec77f788
